test/integration/heartbeat_monitor.py

```diff
@@ -0,0 +1,104 @@
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+'''
+
+import argparse, unittest, sys, os, signal, time
+import requests, json
+from urllib.parse import urlparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--nodepool_pid", help="Nodepool PID")
+parser.add_argument("--prometheus_port", help="CPO prometheus port")
+args = parser.parse_args()
+
+#CPOService_HealthMonitor_heartbeats_sent{shard="1",total_cores="2"} 40
+#CPOService_HealthMonitor_nodepool_down{shard="1",total_cores="2"} 0.000000
+#CPOService_HealthMonitor_nodepool_total{shard="1",total_cores="2"} 1.000000
+
+class TestHeartbeatFailure(unittest.TestCase):
+    def test_heartbeatFailure(self):
+        url = "http://127.0.0.1:" + args.prometheus_port + "/metrics"
+        r = requests.get(url)
+        for line in r.text.splitlines():
+            if "HealthMonitor_heartbeats_sent" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count > 0)
+                print("Heartbeats_sent: ", count)
+            if "HealthMonitor_nodepool_down" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count == 0)
+                print("nodepool_down: ", count)
+            if "HealthMonitor_nodepool_total" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count > 0)
+                print("nodepool_total: ", count)
+
+        os.kill(int(args.nodepool_pid), signal.SIGKILL)
+        time.sleep(2)
+            
+        url = "http://127.0.0.1:" + args.prometheus_port + "/metrics"
+        r = requests.get(url)
+        for line in r.text.splitlines():
+            if "HealthMonitor_heartbeats_sent" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count > 0)
+                print("Heartbeats_sent: ", count)
+            if "HealthMonitor_nodepool_down" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count == 1)
+                print("nodepool_down: ", count)
+            if "HealthMonitor_nodepool_total" in line:
+                count = 0
+                try:
+                    count = int(float(line.split()[1]))
+                except:
+                    continue
+                self.assertTrue(count > 0)
+                print("nodepool_total: ", count)
+
+
+del sys.argv[1:]
```

Comment:
It is needed because the unittest module will throw an exception if it sees any command line arguments it doesn't understand. It doesn't need to be done twice though, I will fix that.